app/src/main/java/com/hedvig/app/feature/embark/EmbarkViewModel.kt

```diff
@@ -103,7 +103,23 @@ abstract class EmbarkViewModel(
         }.flatten()
     }
 
-    fun navigateToPassage(passageName: String) {
+    fun submitAction(nextPassageName: String, submitIndex: Int = 0) {
```

Comment:
Okay so this is where the magic happens ;) I kind of understand, but still having a hard time wrapping my head around embark. So nice to have tests though, good job!

Reply:
That is indeed where the magic happens.

Embark could indeed do with a nice refactor, it's grown a bit hairy. 😬